tests/components/mqtt/test_number.py

```diff
@@ -72,12 +80,29 @@ async def test_run_number_setup(hass, mqtt_mock):
 async def test_run_number_service_optimistic(hass, mqtt_mock):
```

Comment:
Maybe add a non-optimistic version of the test to verify the state is not changed?